File: view/next-project/src/utils/api/purchaseReport.ts
import { PurchaseReport } from '@type/common';

export const post = async (url: string, data: PurchaseReport) => {
  console.log(data);
  const res = await fetch(url, {
    method: 'POST',
    mode: 'cors',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(data),
  }).then((response) => response.json());
  return res;
};

export const put = async (url: string, data: PurchaseReport) => {
  const userID = data.userID;
  const discount = data.discount;
  const addition = data.addition;
  const financeCheck = data.financeCheck;
  const remark = data.remark;
  const buyer = data.buyer;
  const purchaseOrderId = data.purchaseOrderID;
  const putUrl =
    url +
    '?user_id=' +
    userID +
    '&discount=' +
    discount +
    '&addition=' +
    addition +
    '&finance_check=' +
    financeCheck +
    '&remark=' +
    remark +
    '&buyer=' +
    buyer +
    '&purchase_order_id=' +
    purchaseOrderId;
  const res = await fetch(putUrl, {
    method: 'PUT',
    mode: 'cors',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(data),
  }).then((response) => response.json());
  return res;
};

File: api/internals/domain/purchase_report.go
package domain

import (
	"time"
)

type PurchaseReport struct {
	ID              int       `json:"id"`
	UserID          int       `json:"userID"`
	Discount        int       `json:"discount"`
	Addition        int       `json:"addition"`
	FinanceCheck    bool      `json:"financeCheck"`
	PurchaseOrderID int       `json:"purchaseOrderID"`
	Remark          string    `json:"remark"`
	Buyer           string    `json:"buyer"`
	CreatedAt       time.Time `json:"createdAt"`
	UpdatedAt       time.Time `json:"updatedAt"`
}

type PurchaseReportDetails struct {
	PurchaseReport PurchaseReport `json:"purchaseReport"`
	ReportUser     User           `json:"reportUser"`
	PurchaseOrder  PurchaseOrder  `json:"purchaseOrder"`
	OrderUser      User           `json:"orderUser"`
	PurchaseItems  []PurchaseItem `json:"purchaseItems"`
}
